src/wwElement.vue

```diff
@@ -1,78 +1,161 @@
+            wwLib.wwVariable.useComponentVariable({
+                uid: props.uid,
+                name: 'Is Played',
+                type: 'boolean',
+                defaultValue: false,
+            });
+        const { variableValue: currentTimeVariableValue, setValue: setCurrentTimeValue } =
+            wwLib.wwVariable.useComponentVariable({
+                uid: props.uid,
+                name: 'Current time',
+                type: 'number',
+                defaultValue: 0,
+            });
+
+        return { player, isPlayedVariableValue, setIsPlayedValue, currentTimeVariableValue, setCurrentTimeValue };
     },
-    computed: {},
-    watch: {
-        'content.autoplay'(newAuto, oldAuto) {
-            if (this.content.autoplay) {
-                this.$emit('update:content:effect', {
-                    muted: true,
-                });
-            }
-            if (newAuto !== oldAuto && newAuto) {
-                this.updateweWeWebVideo();
-            }
+    computed: {
+        isEditing() {
+            /* wwEditor:start */
+            return this.wwEditorState.editMode === wwLib.wwEditorHelper.EDIT_MODES.EDITION;
+            /* wwEditor:end */
+            // eslint-disable-next-line no-unreachable
+            return false;
         },
-        'content.loop'(newLoop, oldLoop) {
-            if (newLoop !== oldLoop && newLoop) {
-                this.updateweWeWebVideo();
-            }
+        videoId() {
+            if (!this.content.url) return '';
+            return this.content.url.split('m/')[1].split('?')[0];
         },
     },
-    unmounted() {
-        if (this.isEventListener) {
-            const videoEl = document.querySelector('.ww-video-element');
-            videoEl.removeEventListener('click', this.handleVideoClick);
-            this.isEventListener = false;
-        }
+    watch: {
+        isEditing: {
+            async handler() {
+                await this.initPlayer();
+            },
+        },
+        'content.url': {
+            async handler() {
+                await this.initPlayer();
+            },
+        },
+        'content.videoStartTime': {
+            async handler() {
+                await this.initPlayer();
+            },
+        },
+        'content.controls': {
+            async handler() {
+                await this.initPlayer();
+            },
+        },
+        'content.loop'(value) {
+            if (this.player) this.player.setLoop(value);
+        },
+        'content.muted'(value) {
+            if (this.player) this.player.setMuted(value);
+        },
     },
     methods: {
-        getInfoFromUrl(url) {
-            if (!this.content.url) return {};
-            return {
-                id: url.split('m/')[1].split('?')[0],
-                provider: 'vimeo',
-            };
+        async initPlayer() {
+            if (this.player) await this.player.destroy();
+
+            this.$nextTick(async () => {
```

Comment:
plutôt que faire this.$nextTick on peut aussi complétement attendre avec un await no?